nilearn/image/image.py

```diff
@@ -596,3 +599,99 @@ def new_img_like(ref_niimg, data, affine=None, copy_header=False):
+        If given as number, it should be a real number of range between 0 and 100.
+        Both string and real number are used in thresholding_strategy='percentile'.
+
+    thresholding_strategy: string {'percentile', 'img_value'}, \
+        default 'percentile', optional
+        A strategy which takes the value and thresholds the image.
+        If 'percentile', image is thresholded based on the percentage of the
+        score on the image data. The scores which are survived above this
+        percentile are kept.
+        or
+        If 'img_value', voxels which have intensities greater than given float
+        value are kept.
+
+    mask_img: a Nifti like image/object, default is None, optional
+        A mask image used to mask the input image data. For example, if mask image
+        of one particular brain region is given then only those region will be
+        used for processing and rest of the regions are masked out to zero.
+        If None, no masking will be applied. Processing will be done on whole brain.
+
+    Returns
+    -------
+    threshold_img: a Nifti like image/object
+        a thresholded image of the given input image.
+    """
+    from . import resampling
+    from .. import masking
+
+    img = check_niimg(img, atleast_4d=True)
+    img_data = img.get_data()
+    affine = img.get_affine()
+
+    if mask_img is not None:
+        if not _check_same_fov(img, mask_img):
+            mask_img = resampling.resample_img(mask_img, target_affine=affine,
+                                               target_shape=img.shape[:3],
+                                               interpolation="nearest")
+
+        mask_data, _ = masking._load_mask_img(mask_img)
+        # Set as 0 for the values which are outside of the mask
+        img_data[mask_data == 0.] = 0.
+
+    if threshold is not None:
+        if isinstance(threshold, float):
+            # When float value is given to threshold the image directly,
+            # the value will be checking if it is more than maximum.
+            value_check = abs(img_data).max()
+            if abs(threshold) > value_check:
+                raise ValueError("The float value given to threshold directly "
+                                 "must not exceed %d. "
+                                 "You provided threshold=%s " % (value_check,
+                                                                 threshold))
+            cutoff_threshold = threshold
+        elif isinstance(threshold, _basestring) or isinstance(threshold, numbers.Real):
+            if isinstance(threshold, numbers.Number):
+                threshold = ("{0}%").format(threshold)
+            cutoff_threshold = check_threshold(threshold, img_data,
+                                               percentile_calculate=scoreatpercentile,
+                                               name='threshold')
+    else:
+        raise ValueError("The input parameter 'threshold' is empty. "
```

Comment:
Why not handle this option at hte beginning, would that not reduce indentations?